packages/opentelemetry-utils/src/index.ts

```diff
@@ -0,0 +1,59 @@
+import { context, Context, propagation, trace, SpanStatusCode } from '@opentelemetry/api'
+
+/**
+ * Sets attributes to be propagated across child spans under the current active context
+ */
+export const setMultiSpanAttributes = function (attributes: { [key: string]: string }) {
+  const currentBaggage = propagation.getBaggage(context.active())
+  // Create a baggage if there's none
+  let baggage = currentBaggage === undefined ? propagation.createBaggage() : currentBaggage
+  Object.entries(attributes).forEach(([key, value]) => {
+    baggage = baggage.setEntry(key, { value })
+  })
+
+  return propagation.setBaggage(context.active(), baggage)
+}
+
+/**
+ * Add error information to the current active span (if any). Optionally sets the provided attributes on the span too.
+ */
+export const addErrorToActiveSpan = function (error: Error, attributes?: { [key: string]: string }) {
+  const span = trace.getActiveSpan()
+  if (!span) return
+
+  if (attributes !== undefined) {
+    span.setAttributes(attributes)
+  }
+
+  span.recordException(error)
+  span.setStatus({
+    code: SpanStatusCode.ERROR,
+    message: error.message,
+  })
+}
```

Comment:
This is a close to, but more generic version of:
- https://github.com/netlify/build/blob/38149ec7d75e04570da65c58c42ed04f11698491/packages/build/src/tracing/main.ts#L126-L141

The idea being that we don't depend on `ntl/build` error internals but we can create a wrapper around how we set the error state, record the exception and add any attributes to the span. This can then be reused there and in other places.